patches/tModLoader/Terraria/WorldGen.TML.cs

```diff
@@ -7,4 +23,34 @@ public partial class WorldGen
 	private static double _timePass = 0.0; // Used to account for more precise time rates.
 
 	internal static void ClearGenerationPasses() => _generator?._passes.Clear();
+
+	internal static List<GenPass> vanillaGenPasses_internal = new();
+	public static ReadOnlyCollection<GenPass> VanillaGenPasses = vanillaGenPasses_internal.AsReadOnly();
```

Comment:
prefer to return `IReadOnlyCollection`. Sure someone can cast it back to `List` but we're not here stop that.